TICLv5a : Improve skeleton linking and parameters tuning

waredjeb opened the pull request:
This PR improves the current Skeleton Linking
- New Parameters: tuned parameters for hadronic reconstruction.
- Code refactoring: simply code by refactoring pieces
- Modified TICLGraph: modified logic for traversing the graph and build components

- Small changes to TICLDumper: allow to run TICLDumper with TICLv4. It will consume SuperClusters (new with TICLv5) only in TICLv5

New parameters have been included also at HLT 

The performance against TICLv4 can be found in the DPS approval talk https://indico.cern.ch/event/1462666/contributions/6158255/attachments/2940573/5170734/TICLv5_DPS_version3.pdf


Should be tested with a .203 workflow

E.g.
```
test parameters:

    - workflow_opts= -w upgrade
    - workflow = 29896.203, 29691.203, 29691.204
    - enable = profiling, hlt_p2_timing
```

Edit 06/11/2024:

 - `hltTiclTracksterLinks_cfi` and `hltTiclTracksterLinksUnseeded_cfi`,  Both of them were Unseeded with no differences between them, for consistency I removed `hltTiclTracksterLinksUnseeded_cfi` fixing the Egamma sequence accordingly
 - I 've added the L1Seeded module and sequence for the Recovery Tracksters, it is not used but I added it for consistency with `hltTiclTracksterLinksL1Seeded_cfi`

waredjeb committed:
TICLv5.1 : Improve skeleton linking and parameters tuning

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"bdcc39c35e9987eb5bb067a2bcb827a54ea8c0a0":{"files":["HLTrigger/Configuration/python/HLT_75e33/modules/hltTiclTracksterLinksUnseeded_cfi.py","HLTrigger/Configuration/python/HLT_75e33/modules/hltTiclTracksterLinks_cfi.py","RecoHGCal/TICL/plugins/TICLDumper.cc","RecoHGCal/TICL/plugins/TICLGraph.cc","RecoHGCal/TICL/plugins/TICLGraph.h","RecoHGCal/TICL/plugins/TracksterLinkingbySkeletons.cc","RecoHGCal/TICL/plugins/TracksterLinkingbySkeletons.h","RecoHGCal/TICL/python/customiseTICLFromReco.py","RecoHGCal/TICL/python/iterativeTICL_cff.py","RecoHGCal/TICL/python/ticlDumper_cff.py"],"squashed":false,"time":1730198226}},"emoji":{},"last_seen_sha":"bdcc39c35e9987eb5bb067a2bcb827a54ea8c0a0","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46539/42407

cmsbuild commented:
A new Pull Request was created by  @waredjeb for master.

It involves the following packages:

- HLTrigger/Configuration (**hlt**)
- RecoHGCal/TICL (**upgrade, reconstruction**)


@Martin-Grunewald, @Moanwar, @cmsbuild, @jfernan2, @mandrenguyen, @mmusich, @srimanob, @subirsarkar can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @SohamBhattacharya, @VourMa, @apsallid, @felicepantaleo, @forthommel, @hatakeyamak, @lecriste, @missirol, @mmusich, @rovere, @sameasy, @silviodonato, @sobhatta, @youyingli this is something you requested to watch as well.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

felicepantaleo commented:
test parameters:

- workflow_opts= -w upgrade
- workflow = 29896.203
- enable = profiling, hlt_p2_timing

felicepantaleo commented:
@cmsbuild please test

waredjeb commented:
Hi,
I think the tests got stuck

felicepantaleo commented:
@smuzaffar do you have any idea why profiling tests are stuck?

smuzaffar commented:
@felicepantaleo , the node which runs the profiling tests is busy running IB profiling jobs. Let me check if I can improve the situation

cmsbuild commented:
-1

**Failed Tests**: RelVals-INPUT 
**Size**: This PR adds an extra 12KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42504/summary.html
**COMMIT**: bdcc39c35e9987eb5bb067a2bcb827a54ea8c0a0
**CMSSW**: CMSSW_14_2_X_2024-10-30-2300/el8_amd64_gcc12
**Additional Tests**: PROFILING,HLT_P2_TIMING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46539/42504/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **DAS Queries**: The DAS query tests failed, see the summary page for details.


## RelVals-INPUT

- [2024.000001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42504/runTheMatrixINPUT-results/2024.000001_RunJetMET02024D_10k)```2024.000001_RunJetMET02024D_10k/step1_dasquery.log```
- [2024.001001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42504/runTheMatrixINPUT-results/2024.001001_RunZeroBias2024D_10k)```2024.001001_RunZeroBias2024D_10k/step1_dasquery.log```
- [2024.100001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42504/runTheMatrixINPUT-results/2024.100001_RunJetMET02024C_10k)```2024.100001_RunJetMET02024C_10k/step1_dasquery.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [2024.101001](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42504/runTheMatrixINPUT-results/2024.101001_RunZeroBias2024C_10k)
- 2024.000001
- 2024.001001
- 2024.100001
- 2024.101001
</details>



* **HLT P2 Timing**: [chart](https://cmssdt.cern.ch/circles/web/piechart.php?data_name=hlt-p2-timing&resource=time_thread&filter=CMSSW_14_2_X_2024-10-30-2300&dataset=CMSSW_14_2_X_2024-10-30-2300-cmssw46539/el8_amd64_gcc12/36/Phase2Timing_resources)

## Comparison Summary



**Summary**:
 - You potentially [removed 3 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-30-2300+d4760a/65393/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 1056 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-30-2300+d4760a/65393/validateJR.html)
 - DQMHistoTests: Total files compared: 47
 - DQMHistoTests: Total histograms compared: 3668988
 - DQMHistoTests: [Total failures: 789](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-30-2300+d4760a/65393/validateJR.html)
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3668178
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 46 files compared)
 - DQMHistoSizes: changed ( 29896.203 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 205 log files, 175 edm output root files, 47 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-10-30-2300+d4760a/65393/triggerResults): no differences found

jfernan2 commented:
ticlTrackstersMerge is doubling the time, do we understand why? Thanks

https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42504/profiling/29834.21/diff-step3_cpu.resources.json.html

waredjeb commented:
> ticlTrackstersMerge is doubling the time, do we understand why? Thanks
> 
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42504/profiling/29834.21/diff-step3_cpu.resources.json.html

Hi @jfernan2 . I'll be honest, I am quite confused, I am not touching the TracksterMergeProducer with this PR (as you can also see from the changes). ticlTrackstersMerge is part of TICLv4, in this PR I am only touching TICLv5 (workflows .203), so I am not expecting any changes on all the other workflows.

felicepantaleo commented:
I tried running wf `29834.21` (TICLv4) with `CMSSW_14_2_X_2024-10-31-2300` and `CMSSW_14_2_X_2024-10-31-2300+PR46539`, then dumped the step3 in the two configuration and the only difference I see is in `process.ticlTracksterLinks` (TICLv5, not executed in this wf):

```diff
79481c79481,79482
<         cylinder_radius_sqr = cms.vdouble(9, 9),
---
>         cylinder_radius_sqr = cms.vdouble(9, 15),
>         deltaRxy = cms.double(4.0),
79483,79484c79484,79486
<         max_distance_projective_sqr = cms.vdouble(60, 60),
<         max_distance_projective_sqr_closest_points = cms.vdouble(60, 60),
---
>         lower_boundary = cms.vdouble(20, 10),
>         lower_distance_projective_sqr = cms.vdouble(30, 60),
>         lower_distance_projective_sqr_closest_points = cms.vdouble(10, 50),
79486,79488c79488,79490
<         min_distance_z = cms.vdouble(30, 30),
<         min_num_lcs = cms.uint32(7),
<         min_trackster_energy = cms.double(10),
---
>         min_distance_z = cms.vdouble(35, 35),
>         min_num_lcs = cms.uint32(15),
>         min_trackster_energy = cms.double(20),
79492c79494,79497
<         wind = cms.double(0.036)
---
>         upper_boundary = cms.vdouble(150, 100),
>         upper_distance_projective_sqr = cms.vdouble(30, 60),
>         upper_distance_projective_sqr_closest_points = cms.vdouble(5, 30),
>         wind = cms.double(0.072)
79519,79520c79524,79525
<         onnxEnergyModelPath = cms.FileInPath('RecoHGCal/TICL/data/ticlv5/onnx_models/linking/energy_v0.onnx'),
<         onnxPIDModelPath = cms.FileInPath('RecoHGCal/TICL/data/ticlv5/onnx_models/linking/id_v0.onnx'),
---
>         onnxEnergyModelPath = cms.FileInPath('RecoHGCal/TICL/data/ticlv5/onnx_models/patternrecognition/energy_v0.onnx'),
>         onnxPIDModelPath = cms.FileInPath('RecoHGCal/TICL/data/ticlv5/onnx_models/patternrecognition/id_v0.onnx'),
```

mmusich commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 12KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42569/summary.html
**COMMIT**: bdcc39c35e9987eb5bb067a2bcb827a54ea8c0a0
**CMSSW**: CMSSW_14_2_X_2024-11-03-0000/el8_amd64_gcc12
**Additional Tests**: PROFILING,HLT_P2_TIMING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/46539/42569/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **HLT P2 Timing**: [chart](https://cmssdt.cern.ch/circles/web/piechart.php?data_name=hlt-p2-timing&resource=time_thread&filter=CMSSW_14_2_X_2024-11-03-0000&dataset=CMSSW_14_2_X_2024-11-03-0000-cmssw46539/el8_amd64_gcc12/37/Phase2Timing_resources)

## Comparison Summary



**Summary**:
 - You potentially [removed 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-03-0000+d4760a/65441/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 1060 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-03-0000+d4760a/65441/validateJR.html)
 - DQMHistoTests: Total files compared: 47
 - DQMHistoTests: Total histograms compared: 3669183
 - DQMHistoTests: [Total failures: 872](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-03-0000+d4760a/65441/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3668291
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 46 files compared)
 - Checked 205 log files, 175 edm output root files, 47 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-03-0000+d4760a/65441/triggerResults): no differences found

mmusich commented:
+hlt

mmusich commented:
test parameters:
   * enable = hlt_p2_integration

mmusich commented:
@cmsbuild, please test
   * just to be on the safe side

waredjeb committed:
fix HLT configuration for Skeleton Linking, remove Unseeded label

waredjeb committed:
add L1Seeded modules and sequecence for recovery tracksters

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46539/42508

waredjeb commented:
Hi,

I updated the PR with some fixes in the HLT configuration for the Skeleton linking.
On top of that I tried to clean up a bit the TICL part at HLT:
- There was `hltTiclTracksterLinks_cfi` and `hltTiclTracksterLinksUnseeded_cfi`. Both of them were Unseeded with no differences between them, for consistency I removed  `hltTiclTracksterLinksUnseeded_cfi` fixing the Egamma sequence accordingly
- I 've added the L1Seeded module and sequence for the Recovery Tracksters, it is not used but I added it for consistency with `hltTiclTracksterLinksL1Seeded_cfi`

Going to update the PR description as well

cmsbuild commented:
Pull request #46539 was updated. @Martin-Grunewald, @Moanwar, @cmsbuild, @jfernan2, @mandrenguyen, @mmusich, @srimanob, @subirsarkar can you please check and sign again.

mmusich commented:
at the moment it doesn't make sense to run HLT integration or timing tests (broken by cms-sw/cmssw#46582).
We'll have to wait for https://github.com/cms-sw/cmssw/pull/46608 to make it in IBs.

waredjeb commented:
> Hi,
> 
> I updated the PR with some fixes in the HLT configuration for the Skeleton linking. On top of that I tried to clean up a bit the TICL part at HLT:
> 
>     * There was `hltTiclTracksterLinks_cfi` and `hltTiclTracksterLinksUnseeded_cfi`. Both of them were Unseeded with no differences between them, for consistency I removed  `hltTiclTracksterLinksUnseeded_cfi` fixing the Egamma sequence accordingly
> 
>     * I 've added the L1Seeded module and sequence for the Recovery Tracksters, it is not used but I added it for consistency with `hltTiclTracksterLinksL1Seeded_cfi`
> 
> 
> Going to update the PR description as well

To check that the changes to the Egamm sequence are fine, it would be nice to test also with a .204 workflow
E.g:

29896.203, 29691.203, 29691.204

mmusich commented:
test parameters:
   * workflow_opts= -w upgrade
   * workflow = 29896.203, 29691.203, 29691.204
   * enable = hlt_p2_integration, hlt_p2_timing

mmusich commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 12KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42696/summary.html
**COMMIT**: be03f9d0a2a5d3b2e809ad1b8cc9e26c4deac2b0
**CMSSW**: CMSSW_14_2_X_2024-11-08-2300/el8_amd64_gcc12
**Additional Tests**: HLT_P2_INTEGRATION,HLT_P2_TIMING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/46539/42696/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **DAS Queries**: The DAS query tests failed, see the summary page for details.


* **HLT P2 Timing**: [chart](https://cmssdt.cern.ch/circles/web/piechart.php?data_name=hlt-p2-timing&resource=time_thread&filter=CMSSW_14_2_X_2024-11-08-2300&dataset=CMSSW_14_2_X_2024-11-08-2300-cmssw46539/el8_amd64_gcc12/41/Phase2Timing_resources)

## Comparison Summary



**Summary**:
 - You potentially [added 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-08-2300+d4760a/65531/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 1446 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-08-2300+d4760a/65531/validateJR.html)
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3642661
 - DQMHistoTests: [Total failures: 2780](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-08-2300+d4760a/65531/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3639861
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 216 log files, 184 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-08-2300+d4760a/65531/triggerResults): no differences found

mmusich reviewed (commented):
few minor points / questions.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46539/42569

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46539/42569/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46539/42569/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

waredjeb commented on HLTrigger/Configuration/python/HLT_75e33/modules/hltTiclTracksterLinks_cfi.py:
```diff
@@ -6,19 +6,52 @@
-        track_time_quality_threshold = cms.double(0.5),
-        type = cms.string('Skeletons'),
-        wind = cms.double(0.036)
+      cylinder_radius_sqr_split = cms.double(9),
```
Hi Marco! Thanks for the comments. I fixed the code with your suggestions.
Regarding the `PSet`. In principle we would like to be able to have different selections depending on the seeding regions. We still need to study this, but in principle, we would like to have the `PSet`s for the different modules separate so that we can tune the parameters for each module.

mmusich replied:
That's fine of course, but then you could still reference two different global `PSets` when that becomes necessary (no strong push for that,  just it looks a tiny bit less error prone)

waredjeb replied:
Hi @mmusich, before committing, I wanted to double-check with you if what I did makes sense.
I created a PSet `hltTiclTracksterLinksPSet_cfi.py` (not sure about the naming of the file, I couldn't spot a general convention from the other files) in `HLTrigger/Configuration/python/HLT_75e33/psets` 

I am then importing this PSet in `HLTrigger/Configuration/python/HLT_75e33/modules`

```diff
diff --git a/HLTrigger/Configuration/python/HLT_75e33/modules/hltTiclTracksterLinksL1Seeded_cfi.py b/HLTrigger/Configuration/python/HLT_75e33/modules/hltTiclTracksterLinksL1Seeded_cfi.py
index a5c9f73b188..d89b5ea6233 100644
--- a/HLTrigger/Configuration/python/HLT_75e33/modules/hltTiclTracksterLinksL1Seeded_cfi.py
+++ b/HLTrigger/Configuration/python/HLT_75e33/modules/hltTiclTracksterLinksL1Seeded_cfi.py
@@ -1,58 +1,12 @@
 import FWCore.ParameterSet.Config as cms
+from ..psets.hltTiclTracksterLinksPSet_cfi import hltTiclTracksterLinksPSet 
 
 hltTiclTracksterLinksL1Seeded = cms.EDProducer("TracksterLinksProducer",
     detector = cms.string('HGCAL'),
     layer_clusters = cms.InputTag("hltHgcalMergeLayerClustersL1Seeded"),
     layer_clustersTime = cms.InputTag("hltHgcalMergeLayerClustersL1Seeded","timeLayerCluster"),
     inferenceAlgo = cms.string('TracksterInferenceByDNN'),
-    linkingPSet = cms.PSet(
-      cylinder_radius_sqr_split = cms.double(9),
-      proj_distance_split = cms.double(5),
-      track_time_quality_threshold = cms.double(0.5),
-      min_num_lcs = cms.uint32(15),
-      min_trackster_energy = cms.double(20),
-      pca_quality_th = cms.double(0.85),
-      dot_prod_th = cms.double(0.97),
-      deltaRxy = cms.double(4),
-      lower_boundary = cms.vdouble(
-        20,
-        10
-      ),
-      upper_boundary = cms.vdouble(
-        150,
-        100
-      ),
-      upper_distance_projective_sqr = cms.vdouble(
-        30,
-        60
-      ),
-      lower_distance_projective_sqr = cms.vdouble(
-        30,
-        60
-      ),
-      min_distance_z = cms.vdouble(
-        35,
-        35
-      ),
-      upper_distance_projective_sqr_closest_points = cms.vdouble(
-        5,
-        30
-      ),
-      lower_distance_projective_sqr_closest_points = cms.vdouble(
-        10,
-        50
-      ),
-      max_z_distance_closest_points = cms.vdouble(
-        35,
-        35
-      ),
-      cylinder_radius_sqr = cms.vdouble(
-        9,
-        15
-      ),
-      algo_verbosity = cms.int32(0),
-      type = cms.string('Skeletons')
-    ),
+    linkingPSet = hltTiclTracksterLinksPSet,
     pluginInferenceAlgoTracksterInferenceByDNN = cms.PSet(
         algo_verbosity = cms.int32(0),
         onnxPIDModelPath = cms.FileInPath('RecoHGCal/TICL/data/ticlv5/onnx_models/linking/id_v0.onnx'),
```

And same thing for the L1Seeded version of the module. Does it look reasonable to you?

mmusich replied:
>  Does it look reasonable to you?

yes, it does!

cmsbuild commented:
Pull request #46539 was updated. @Martin-Grunewald, @Moanwar, @cmsbuild, @jfernan2, @mandrenguyen, @mmusich, @srimanob, @subirsarkar can you please check and sign again.

mmusich commented:
@cmsbuild, please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 12KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42775/summary.html
**COMMIT**: 75b3afc51ab153cbf1b143dfcee1bf476cf28cc7
**CMSSW**: CMSSW_14_2_X_2024-11-12-1100/el8_amd64_gcc12
**Additional Tests**: HLT_P2_INTEGRATION,HLT_P2_TIMING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46539/42775/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestDQMGUIUpload had ERRORS
</pre>

* **HLT P2 Timing**: [chart](https://cmssdt.cern.ch/circles/web/piechart.php?data_name=hlt-p2-timing&resource=time_thread&filter=CMSSW_14_2_X_2024-11-12-1100&dataset=CMSSW_14_2_X_2024-11-12-1100-cmssw46539/el8_amd64_gcc12/43/Phase2Timing_resources)

## Comparison Summary



There are some workflows for which there are errors in the baseline:
 2024.101001 step 1
2024.202001 step 1
2024.303001 step 1
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

**Summary**:
 - You potentially [removed 4 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-12-1100+d4760a/65597/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 1776 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-12-1100+d4760a/65597/validateJR.html)
 - DQMHistoTests: Total files compared: 46
 - DQMHistoTests: Total histograms compared: 3369424
 - DQMHistoTests: [Total failures: 2761](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-12-1100+d4760a/65597/validateJR.html)
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3366642
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 45 files compared)
 - DQMHistoSizes: changed ( 29896.203 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 204 log files, 172 edm output root files, 46 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-12-1100+d4760a/65597/triggerResults): no differences found

mmusich commented:
> I found 1 errors in the following unit tests:

errors are unrelated, see https://github.com/cms-sw/cmssw/issues/46682

waredjeb committed:
add GlobalPSet for Linking plugin

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-46539/42629

cmsbuild commented:
Pull request #46539 was updated. @Martin-Grunewald, @Moanwar, @cmsbuild, @jfernan2, @mandrenguyen, @mmusich, @srimanob, @subirsarkar can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 40KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42800/summary.html
**COMMIT**: d112936fcfae380c526fd207013d5cdf5b4606f1
**CMSSW**: CMSSW_14_2_X_2024-11-12-2300/el8_amd64_gcc12
**Additional Tests**: HLT_P2_INTEGRATION,HLT_P2_TIMING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46539/42800/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestDQMGUIUpload had ERRORS
</pre>

* **HLT P2 Timing**: [chart](https://cmssdt.cern.ch/circles/web/piechart.php?data_name=hlt-p2-timing&resource=time_thread&filter=CMSSW_14_2_X_2024-11-12-2300&dataset=CMSSW_14_2_X_2024-11-12-2300-cmssw46539/el8_amd64_gcc12/44/Phase2Timing_resources)

## Comparison Summary



**Summary**:
 - You potentially [removed 5 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-12-2300+d4760a/65610/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 1781 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-12-2300+d4760a/65610/validateJR.html)
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3642661
 - DQMHistoTests: [Total failures: 2821](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-12-2300+d4760a/65610/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3639820
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 216 log files, 184 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-12-2300+d4760a/65610/triggerResults): no differences found

mmusich commented:
@cmsbuild, please test
   * let's try again

cmsbuild commented:
+1

**Size**: This PR adds an extra 12KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d4760a/42875/summary.html
**COMMIT**: d112936fcfae380c526fd207013d5cdf5b4606f1
**CMSSW**: CMSSW_14_2_X_2024-11-14-2300/el8_amd64_gcc12
**Additional Tests**: HLT_P2_INTEGRATION,HLT_P2_TIMING
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46539/42875/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **DAS Queries**: The DAS query tests failed, see the summary page for details.


* **HLT P2 Timing**: [chart](https://cmssdt.cern.ch/circles/web/piechart.php?data_name=hlt-p2-timing&resource=time_thread&filter=CMSSW_14_2_X_2024-11-14-2300&dataset=CMSSW_14_2_X_2024-11-14-2300-cmssw46539/el8_amd64_gcc12/46/Phase2Timing_resources)

## Comparison Summary



**Summary**:
 - You potentially [added 4 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-14-2300+d4760a/65666/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 1779 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-14-2300+d4760a/65666/validateJR.html)
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3642837
 - DQMHistoTests: [Total failures: 2729](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-14-2300+d4760a/65666/validateJR.html)
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3640087
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.004 KiB( 48 files compared)
 - DQMHistoSizes: changed ( 29896.203 ): 0.004 KiB        MessageLogger/Warnings
 - Checked 216 log files, 184 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_2_X_2024-11-14-2300+d4760a/65666/triggerResults): no differences found

jfernan2 commented:
+1

cmsbuild commented:
Milestone for this pull request has been moved to CMSSW_15_0_X. Please open a backport if it should also go in to CMSSW_14_2_X.

Moanwar commented:
+Upgrade

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @sextonkennedy, @rappoccio, @antoniovilela, @mandrenguyen (and backports should be raised in the release meeting by the corresponding L2)

mandrenguyen commented:
+1
